Error when the field `AdditionalProperties: true` is set.

I am trying to generate documents for our crds with your great tool. All in all it works fine, but if our crd contains the `AdditionalProperties: true` field, an error occurs and no documents are generated.

A simple example crd looks like this:

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: example.examples.com
spec:
  group: examples.com
  names:
    kind: example
  versions:
  - name: v1
    served: true
    storage: true
    schema:
      openAPIV3Schema:
        type: object
        properties:
          spec:
            type: object
            properties:
               values:
                  description: test description
                  type: object
                  additionalProperties: true # Error field
                  default: {}
  scope: Namespaced
```
Trying to create documents with `crdoc --resources=./ --output=./test.md` results in the following error message: 

```sh
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1195edc]

goroutine 1 [running]:
github.com/mesh-for-data/crdoc/pkg/builder.getTypeName(0xc0004f2200, 0xc00023de00, 0x0)
	/home/wetzel/go/pkg/mod/github.com/mesh-for-data/crdoc@v0.3.0/pkg/builder/builder.go:223 +0xdc
github.com/mesh-for-data/crdoc/pkg/builder.(*ModelBuilder).addTypeModels(0xc000213d30, 0xc000497f40, 0xc0004f0090, 0xc0001cb9c8, 0x13, 0xc0004f2200, 0x0, 0x0, 0xc0004ee0e0, 0x10)
	/home/wetzel/go/pkg/mod/github.com/mesh-for-data/crdoc@v0.3.0/pkg/builder/builder.go:140 +0x45
github.com/mesh-for-data/crdoc/pkg/builder.(*ModelBuilder).addTypeModels(0xc000213d30, 0xc000497f40, 0xc0004f0090, 0xc0004ee150, 0xc, 0xc0004f2000, 0x0, 0x0, 0x0, 0x0)
	/home/wetzel/go/pkg/mod/github.com/mesh-for-data/crdoc@v0.3.0/pkg/builder/builder.go:156 +0x3ba
github.com/mesh-for-data/crdoc/pkg/builder.(*ModelBuilder).addTypeModels(0xc000213d30, 0xc000497f40, 0xc0004f0090, 0xc0004ee080, 0x7, 0xc00023d600, 0x1, 0x1, 0x1d0cea0, 0x0)
	/home/wetzel/go/pkg/mod/github.com/mesh-for-data/crdoc@v0.3.0/pkg/builder/builder.go:156 +0x3ba
github.com/mesh-for-data/crdoc/pkg/builder.(*ModelBuilder).Add(0xc000213d30, 0xc0004982c0, 0xc0004ac970, 0x1)
	/home/wetzel/go/pkg/mod/github.com/mesh-for-data/crdoc@v0.3.0/pkg/builder/builder.go:93 +0x2a6
main.RootCmd.func2(0xc000394c80, 0xc0003fa4e0, 0x0, 0x2, 0x0, 0x0)
	/home/wetzel/go/pkg/mod/github.com/mesh-for-data/crdoc@v0.3.0/main.go:69 +0x274
github.com/spf13/cobra.(*Command).execute(0xc000394c80, 0xc000194160, 0x2, 0x2, 0xc000394c80, 0xc000194160)
	/home/wetzel/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc000394c80, 0xc000000180, 0xc000213f78, 0x408025)
	/home/wetzel/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/wetzel/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
	/home/wetzel/go/pkg/mod/github.com/mesh-for-data/crdoc@v0.3.0/main.go:103 +0x2b
```

If I remove the `AdditionalProperties: true` field, it works fine again.  Any idea what could be the problem?

--- a/example/crds/example.yaml
+++ b/example/crds/example.yaml
@@ -18,6 +18,11 @@ spec:
     schema:
       openAPIV3Schema:
         properties:
+          arbitrary:
+            description: arbitrary field
+            type: object
+            additionalProperties: true
+            default: {}
           messages:
             items:
               additionalProperties:

--- a/example/output.md
+++ b/example/output.md
@@ -50,6 +50,15 @@ Resource Types:
       <td>Refer to the Kubernetes API documentation for the fields of the `metadata` field.</td>
       <td>true</td>
       </tr><tr>
+        <td><b>arbitrary</b></td>
+        <td>map[string]string</td>
+        <td>
+          arbitrary field<br/>
+          <br/>
+            <i>Default</i>: map[]<br/>
+        </td>
+        <td>false</td>
+      </tr><tr>
         <td><b><a href="#examplemessagesindexkey">messages</a></b></td>
         <td>[]map[string]object</td>
         <td>

--- a/pkg/builder/builder.go
+++ b/pkg/builder/builder.go
@@ -220,7 +220,7 @@ func getTypeName(props *apiextensions.JSONSchemaProps) string {
 
 	// map
 	if props.Type == "object" && props.AdditionalProperties != nil {
-		if props.AdditionalProperties.Schema == nil && props.AdditionalItems.Allows {
+		if props.AdditionalProperties.Schema == nil && props.AdditionalProperties.Allows {
 			return "map[string]string"
 		}
 		return "map[string]"

